### PR TITLE
More common cleanups

### PR DESCRIPTION
**Pull request by hifi**

### CODE REVIEW

File: common/palette.cpp [hunk 1]
```diff
@@ -234,7 +220,7 @@ void Fade_Palette_To(void* palette1, unsigned int delay, void (*callback)())
  *   04/27/1994 BR : Converted to 32-bit                                   *
  *   08/02/1994 SKB : Made private                                         *
  *=========================================================================*/
-PRIVATE void __cdecl Determine_Bump_Rate(void* palette, int delay, short* ticks, short* rate)
+PRIVATE void Determine_Bump_Rate(void* palette, int delay, short* ticks, short* rate)
```
Comment (wdigger) [inline common/palette.cpp]: PRIVATE?

File: common/palette.cpp [hunk 2]
```diff
@@ -41,11 +41,11 @@
 /*
 ********************************* Includes **********************************
 */
-//#include <mem.h>
 
 #include "palette.h"
 #include "timer.h"
 #include "wwstd.h"
```
Comment (wdigger) [inline common/palette.cpp]: I propose to remove 'wwstd.h' and include necessary headers directly, it's helps to understand what windows specific are actually used.

File: common/gbuffer.h [hunk 3]
```diff
@@ -115,6 +115,7 @@
 #include "drawbuff.h"
 
 #include <cstdint>
+#include <cstdio>
```
Comment (wdigger) [inline common/gbuffer.h]: @OmniBlade was strongly against using С++ headers without 'using std' or 'std::' for all std lib calls.

Comment (OmniBlade) [reply common/gbuffer.h]: Indeed, it causes portability issues with some implementations of the C++ standard library as they are not required to export the symbols in the global namespace from the C++ versions of those headers. Recent GCC and libstdc++ headers don't for example and neither do the libc++/clang headers IIRC. Options are either to just use the C headers which do ensure the symbols are global namespace, use `std::` everywhere or do a `using std::` for each function used in the translation unit.

**Comment by hifi** [comment]
FYI this currently also crashes Remaster TD so marked as draft so it's not accidentally approved and merged even after fixes.

File: common/gbuffer.h (hunk 3 above)
Comment (hifi) [reply common/gbuffer.h]: Changed to C headers.

File: common/palette.cpp (hunk 2 above)
Comment (hifi) [reply common/palette.cpp]: Removed wwstd compeltely as it was redundant.

**Comment by hifi** [comment]
I fixed the TD remaster crash and tried to fix all PRIVATE removals back to static. This is a good bunch of cleanups and it makes most of common compilable on Linux.

File: common/mouseww.cpp [hunk 4]
```diff
@@ -85,13 +85,17 @@ WWMouseClass::WWMouseClass(GraphicViewPortClass* scr, int mouse_max_width, int m
...
     EraseBuffHotY = -1;
-    EraseFlags = FALSE;
+    EraseFlags = false;
 
     _Mouse = this;
 
+#ifdef _WIN32
     // Add TIME_KILL_SYNCHRONOUS flag. ST - 2/13/2019 5:07PM
     // TimerHandle = timeSetEvent( 1000/60 , 1 , ::Process_Mouse, 0 , TIME_PERIODIC);
-#ifndef REMASTER_BUILD
```
Comment (OmniBlade) [inline common/mouseww.cpp:94]: Are we sure we are fine ungating this from the vanilla builds only? I assume that the remaster never creates this class or something?

**Review by OmniBlade — APPROVED** [review]
Approved, just need clarifying if the ungating from vanilla was intentional for the WWMouseClass constructor.

File: common/mouseww.cpp (hunk 4 above)
Comment (hifi) [reply common/mouseww.cpp:94]: Cleaned up the gates to be more precise. As long as it compiles on Linux right now I'm content.

**Pull request merged**